packages/block-library/src/navigation/edit.js

```diff
@@ -126,56 +182,181 @@ function Navigation( {
 	}
 
 	function handleCreateFromExistingPages() {
-		updateNavItemBlocks( defaultPagesNavigationItems );
+		updateNavItemBlocks( navLinkBlocksFromPages );
 		selectBlock( clientId );
 	}
 
-	const hasPages = hasResolvedPages && pages && pages.length;
+	function handleCreateFromExistingMenu() {
+		updateNavItemBlocks( navLinkBlocksFromMenuItems );
+		selectBlock( clientId );
+	}
 
-	const blockInlineStyles = {
-		fontSize: fontSize.size ? fontSize.size + 'px' : undefined,
-	};
+	function handleCreate() {
+		const { key } = selectedDropDownOption;
+
+		// Explicity request to create empty.
+		if ( key === CREATE_EMPTY_OPTION_VALUE ) {
+			return handleCreateEmpty();
+		}
+
+		// Create from Pages.
+		if ( hasPages && key === CREATE_FROM_PAGES_OPTION_VALUE ) {
+			return handleCreateFromExistingPages();
+		}
+
+		// Create from WP Menu (if exists and not empty).
+		if (
+			hasMenus &&
+			selectedDropDownOption &&
+			navLinkBlocksFromMenuItems?.length
+		) {
+			return handleCreateFromExistingMenu();
+		}
+
+		// Default to empty menu
+		return handleCreateEmpty();
+	}
+
+	function buildPlaceHolderInstructionText() {
+		if ( hasMenus && hasPages ) {
+			return __(
+				'Create a Navigation from all existing pages, or chose a Menu.'
+			);
+		}
+
+		if ( ! hasMenus && ! hasPages ) {
+			return __( 'Create a an empty Navigation.' );
+		}
+
+		if ( hasMenus && ! hasPages ) {
+			return __( 'Create a Navigation from a Menu or create empty.' );
+		}
+
+		if ( ! hasMenus && hasPages ) {
+			return __(
+				'Create a Navigation from all existing pages, or create empty.'
+			);
+		}
+	}
+
+	const dropDownOptions = [
+		{
+			id: CREATE_PLACEHOLDER_VALUE,
+			name: __( 'Select where to start from…' ),
+		},
+		...( hasMenus ? menus : [] ),
```

Comment:
Not sure if the CustomSelectControl supports aria-labels for options, but I noticed voiceover simply reads the name of the menu. It might be good to augment the option with some more information for screenreader users, e.g. "Create a navigation using menu 'Main Menu'".

Reply:
It doesn't support it. This feels like a good candidate for a followup Issue with a complementary Issue for enhancing the `CustomSelectControl`. Would you agree?

Reply:
I've looked into this further and `aria-label` should only be used when there is no visible label text available. IN this case we have visible label text which is the option's text itself. 

An alternative is `aria-describedby` which provides a reference to a location which can provide additional context. If we enhance `CustomSelectControl` to allow this attr we can have an element which reacts to the selection providing additional context to users of assistive technologies.

Here is an Issue which tracks this:
https://github.com/WordPress/gutenberg/issues/22988

Reply:
Jerry asked about labeling in the Slack accessibility channel. https://wordpress.slack.com/archives/C02RP4X03/p1591390740446600

Here is a reply from Joe Dolson.
"Using an option for a label introduces a few problems: it’s disabled, so the color contrast is a problem. Disabled controls are an exception to color contrast guidelines, but labels aren’t. 2nd, (which may not be relevant in this use case), the label would be non-obvious if a value is selected in the select, since it would no longer be the first perceived value, and 3rd, screen readers will skip the disabled option when focusing on the control, and go directly to the first enabled option. The faux label would not be read. If the option is also given the “selected” value, then it will be focused by a screen reader, but will also be described as an invalid entry, which could be confusing."

Reply:
@paaljoachim @jeryj Here is [a followup Issue](https://github.com/WordPress/gutenberg/issues/22999) where the most appropriate solution an be discussed and implemented.